commit 154b7f3f537421a35493072c8bc7576883427db8 (HEAD)
Author: Keno Fischer <keno@juliacomputing.com>
Date:   Tue Jun 13 16:17:38 2023 +0000

    Fix build on 32-bit platforms
    
    `unsigned long` is 32bit on many 32bit platforms.
    Use `unsigned long long` instead.

diff --git a/util/compiler.cpp b/util/compiler.cpp
index 2cda14c..49cb9af 100644
--- a/util/compiler.cpp
+++ b/util/compiler.cpp
@@ -44,15 +44,15 @@ unsigned num_sign_bits(uint64_t n) {
 }
 
 uint64_t add_saturate(uint64_t a, uint64_t b) {
-  unsigned long res;
+  unsigned long long res;
   static_assert(sizeof(res) == sizeof(uint64_t));
-  return __builtin_uaddl_overflow(a, b, &res) ? UINT64_MAX : res;
+  return __builtin_uaddll_overflow(a, b, &res) ? UINT64_MAX : res;
 }
 
 uint64_t mul_saturate(uint64_t a, uint64_t b) {
-  unsigned long res;
+  unsigned long long res;
   static_assert(sizeof(res) == sizeof(uint64_t));
-  return __builtin_umull_overflow(a, b, &res) ? UINT64_MAX : res;
+  return __builtin_umulll_overflow(a, b, &res) ? UINT64_MAX : res;
 }
 
 uint64_t divide_up(uint64_t n, uint64_t amount) {

